### PR TITLE
Working Branch for Enumerable Methods

### PR DESCRIPTION
**Pull request by Amita-Roy**
Created followed methods

#my_each, a method that is identical to #each but (obviously) does not use #each and returns the same thing as #each as well.
 #my_each_with_index in the same way.
#my_select
 #my_all?
#my_any?
#my_none?
#my_count
#my_map
#my_inject
 #my_inject
 #multiply_els which multiplies all the elements of the array together by using #my_inject
#my_map

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,142 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    for e in self
+      yield e
+    end
+  end
+
+  def my_each_with_index
```
Comment (davitomix) [inline main.rb:10]: Screen Shot 2020-08-08 at 12 29 46 PM
****
Screen Shot 2020-08-08 at 12 30 14 PM
****
Screen Shot 2020-08-08 at 12 30 41 PM

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+    if arg.nil? && block_given?
+      my_each do |element|
+        return false unless yield element
+      end
+    end
+    return !(self.include?(nil) || self.include?(false))
+  end
+
+  def my_any?(arg = nil)
```
Comment (davitomix) [inline main.rb:48]: Screen Shot 2020-08-08 at 12 31 26 PM

File: main.rb [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    new_arr = []
+    return enum_for unless proc || block_given?
+
+    for e in self
+      new_arr << (proc ? proc.call(e) : yield(e))
+    end
+    return new_arr
+  end
+
+  def my_inject(initial = nil, sym = nil)
```
Comment (davitomix) [inline main.rb:119]: Screen Shot 2020-08-08 at 12 31 44 PM

**Comment by lekegitrepo** [comment]
#### Hi @Amita-Roy and @Haywhizzz :clap: :clap:
#### You have done a good :+1: implementing the required changes.

#### This project is approved.

#### _Reviewed by Temitayo Adeleke._

**Pull request merged**